core/services/bulletprooftxmanager/eth_broadcaster_test.go

```diff
@@ -227,6 +232,71 @@ func TestEthBroadcaster_ProcessUnstartedEthTxs_Success(t *testing.T) {
 
 		ethClient.AssertExpectations(t)
 	})
+
+	t.Run("sends transactions with type 0x2 in EIP-1559 mode", func(t *testing.T) {
+		cfg.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(true)
+		rnd := int64(1000000000 + rand.Intn(5000))
+		cfg.Overrides.GlobalEvmGasTipCapDefault = big.NewInt(rnd)
+		cfg.Overrides.GlobalEvmMaxGasPriceWei = big.NewInt(rnd + 1)
+
+		eipTxWithoutAl := bulletprooftxmanager.EthTx{
+			FromAddress:    fromAddress,
+			ToAddress:      toAddress,
+			EncodedPayload: []byte{42, 0, 0},
+			Value:          assets.NewEthValue(142),
+			GasLimit:       gasLimit,
+			CreatedAt:      time.Unix(0, 0),
+			State:          bulletprooftxmanager.EthTxUnstarted,
+		}
+		eipTxWithAl := bulletprooftxmanager.EthTx{
+			FromAddress:    fromAddress,
+			ToAddress:      toAddress,
+			EncodedPayload: []byte{42, 42, 0},
+			Value:          assets.NewEthValue(242),
+			GasLimit:       gasLimit,
+			CreatedAt:      time.Unix(0, 1),
+			State:          bulletprooftxmanager.EthTxUnstarted,
+			AccessList:     bulletprooftxmanager.NullableEIP2930AccessListFrom(gethTypes.AccessList{gethTypes.AccessTuple{Address: cltest.NewAddress(), StorageKeys: []gethCommon.Hash{utils.NewHash()}}}),
+		}
+		ethClient.On("SendTransaction", mock.Anything, mock.MatchedBy(func(tx *gethTypes.Transaction) bool {
+			return tx.Nonce() == uint64(3) && tx.Value().Cmp(big.NewInt(142)) == 0
+		})).Return(nil).Once()
+		ethClient.On("SendTransaction", mock.Anything, mock.MatchedBy(func(tx *gethTypes.Transaction) bool {
+			return tx.Nonce() == uint64(4) && tx.Value().Cmp(big.NewInt(242)) == 0
+		})).Return(nil).Once()
+
+		require.NoError(t, db.Save(&eipTxWithAl).Error)
+		require.NoError(t, db.Save(&eipTxWithoutAl).Error)
+
+		// Do the thing
+		require.NoError(t, eb.ProcessUnstartedEthTxs(keyState))
+
+		// Check eipTxWithAl and it's attempt
+		// This was the earlier one sent so it has the lower nonce
+		eipTxWithAl, err := cltest.FindEthTxWithAttempts(db, eipTxWithAl.ID)
+		require.NoError(t, err)
+		assert.False(t, eipTxWithAl.Error.Valid)
```

Comment:
I tried to do this and ran into numerous annoyances, for example comparing BroadcastAt doesn't work (firstly because we dont care what it is as long as its not nil, secondly because timestamps lose precision when persisted). Other ones are things like comparing pointers vs. the exact values etc, in the end it didn't add much.